src/server/index.ts

```diff
@@ -0,0 +1,58 @@
+import e from 'express';
+import morgan from 'morgan';
+import http from 'http';
+import signale from 'signale';
+import passport from 'passport';
+import bodyParser from 'body-parser';
+import { Strategy as LocalStrategy } from 'passport-local';
+import { authorize } from '../passport';
+import { home, greet, dbInit, search, filter } from '../routes';
+import { DbClient } from '../dal/interface/index';
+
+/**
+ * Encapsulates the server configuration and routes logic.
+ */
+export class Server {
+    private client: DbClient;
+    private app: e.Application;
+    private port: string | number;
+
+    constructor() {
+        this.app = e();
+        this.port = process.env.PORT as string || 3000 as number;
+        this.configure();
+        this.routes();
+        this.client = new DbClient({
+            url: 'mongodb://mongo',
+            port: 27017,
+            dbName: 'admin'
```

Comment:
We should create a new database, specific to Content Specification Explorer. I'm aware that it means we'll need additional logic to check for its existence, but I think it's safer and less confusing than using the shipped `admin` database.

Reply:
agreed, this is just carry over code from trying to insert the data.